packages/utils/src/asAbortable.ts

```diff
@@ -22,23 +22,20 @@ export function asAbortable<T>(
     abortSignal?: AbortSignal,
     customErrorContext?: string
 ): Promise<T> {
-    if (abortSignal?.aborted === true) {
+    if (abortSignal === undefined) {
+        return promise
+    }
+    if (abortSignal.aborted) {
         return Promise.reject(new AbortError(customErrorContext))
     }
     let abortListener: () => void
     return new Promise<T>((resolve, reject) => {
-        if (abortSignal !== undefined) {
-            abortListener = () => {
-                reject(new AbortError(customErrorContext))
-            }
-            // TODO remove the type casting when type definition for abortController has been updated to include addEventListener
-            (abortSignal as any).addEventListener('abort', abortListener)
+        abortListener = () => {
+            reject(new AbortError(customErrorContext))
         }
+        abortSignal.addEventListener('abort', abortListener)
```

Comment:
In the `finally` block, the event listener gets removed in any case. We could add `{ once: true }` but that would mean that in the case of an abort signal occurring (before `promise` resolves), the `abortListener` would already been removed when we arrive at the `finally` block. Removing a non-existing listener is no-op so it probably wouldn't really matter. So a matter of taste I guess. It would be a bit more "playing it safe" to add the `{ once: true }` but OTOH, it's not really needed in this case.